Spring base jp

hjp1016 opened the pull request:
still working

NotableDeveloper committed:
empty branch init

NotableDeveloper committed:
Create README.md

hjp1016 committed:
init SpringBase

hjp1016 committed:
feat : BookController 구현

- 도서 검색 기능 구현 완료
- 도서 대여 기능 구현 예정

hjp1016 committed:
feat : BookController 수정 및 추가

- 책 대여하기 추가
- 책 반납하기 추가

hjp1016 committed:
feat : BookController 수정

- ResponsBody 삭제
- 조회 접근 url PathVariable로 수정

hjp1016 committed:
feat : JdbcBookRepository 추가

- JdbcBookRepostiory 생성

hjp1016 committed:
feat : JdbcBookRepository 수정

- 데이터베이스 환경변수 설정
- 구현할 함수 선언

hjp1016 committed:
feat : JdbcBookRepository 수정

- 책 추가기능 구현
- 책 제목,  ISBN, 작가, 카테고리 조회 기능 구현
- 책 전체 조회 구현

hjp1016 committed:
feat : book save 수정

- 데이터 추가 컬럼 수정
- 대여, 반납 시스템 진행중

NotableDeveloper commented on src/main/java/com/example/demo/repository/JdbcBookRepository.java:
```diff
@@ -75,6 +73,20 @@ public Book findByISBN(String isbn) {
   public boolean existsByIsbn(String isbn) {
+
+    try {
+      String sql = "select * from book where isbn=?";
```
SQL의 exists 구문을 사용하면 아래에서 count 변수를 사용하지 않아도 될 것으로 보입니다.

NotableDeveloper commented on src/main/java/com/example/demo/repository/JdbcBookRepository.java:
```diff
@@ -181,10 +193,47 @@ public List<Book> findByCategory(BookCategory category){
   @Override
   public Book borrowBook(String isbn){
+    try {
+      String sql = "update book set book_status = 'BORROWING' where isbn=? ";
```
book_status 에 BORROWING을 하드 코딩하지 않고, PreparedStatement의 장점을 살려서 값을 추가하는 방식으로 수정 바랍니다.

NotableDeveloper commented on src/main/java/com/example/demo/repository/JdbcBookRepository.java:
```diff
@@ -181,10 +193,47 @@ public List<Book> findByCategory(BookCategory category){
   @Override
   public Book returnBook(String isbn){
+    try {
+      String sql = "update book set book_status = 'AVAILABLE' where isbn=? ";
```
book_status 에 AVAILABLE을 하드 코딩하지 않고, PreparedStatement의 장점을 살려서 값을 추가하는 방식으로 수정 바랍니다.

NotableDeveloper commented on src/main/java/com/example/demo/repository/JdbcBookRepository.java:
```diff
@@ -181,10 +193,47 @@ public List<Book> findByCategory(BookCategory category){
+    }catch(SQLException e){
+      e.printStackTrace();
+    }
     return null;
```
메서드의 반환은 null이 되어서는 안됩니다.

NotableDeveloper commented on src/main/java/com/example/demo/repository/JdbcBookRepository.java:
```diff
@@ -181,10 +193,47 @@ public List<Book> findByCategory(BookCategory category){
+    }catch(SQLException e){
+      e.printStackTrace();
+    }
     return null;
```
메서드의 반환은 null이 되어서는 안됩니다.

hjp1016 committed:
chore : JPA 의존성 추가

hjp1016 committed:
feat : JpaRepository 추가
- JpaBookRepositroy 추가
- BookService에서 JpaRepositroy 타입을 주입받도록 수정
- JdbcRepository 생성자 주석처리

hjp1016 committed:
chore : application.properties 내용 추가
- application.properties에 db 접속관련 내용추가
- 접속을 위한 환경변수 추가

hjp1016 committed:
feat : Book 클래스 수정
- Entity 어노테이션 추가
- 기본키를 사용하는 id 값 추가

hjp1016 committed:
feat : findByIsbn 기능추가
- JPA를 이용한 ISBN으로 도서 조회하기
- BookService에서 사용하는 매서드를 findByIsbn으로 수정

hjp1016 committed:
feat : findByTitleContains 구현
- 제목을 포함하는 도서 찾기 구현
- 생성한 Repository를 service에서 사용

hjp1016 committed:
feat : findByAuthorContains 추가
- 저자로 도서 검색하기 기능 추가
- 추가한 메서드를 Service에서 사용

hjp1016 committed:
feat : findAllByCategory 구현
- findAllByCategory 기능 구현
- service에서  추가한 메서드를 사용하도록 수정

hjp1016 committed:
feat : enum 데이터 형식 추가
- enum으로 인식 할 수 있도록 어노테이션 추가
- table을 create에서 update로 변경

hjp1016 committed:
feat : 도서 대여 반납 기능 완료
- Transactionl 어노테이션 추가
- 도서 대여 반납 기능 구현
